src/AppBundle/Entity/Client.php

```diff
@@ -697,10 +695,25 @@ public function setClientContacts($clientContacts)
     }
 
     /**
-     * @return Boolean
+     * @return array $coDeps an array of users sorted by firstname, or email if no firstname
      */
-    public function isMultiDeputy()
+    public function getCoDeputies()
     {
-        return (is_array($this->users) && count($this->users) > 0);
+        $coDeps = [];
+        if (is_array($this->users) && count($this->users) > 0) {
+            foreach ($this->users as $user) {
+                if (!$user->getFirstname()) {
+                    $matches = [];
+                    preg_match('(^\w+)', $user->getEmail(), $matches);
+                    if (!empty($matches[0])) {
+                        $coDeps[strToLower($matches[0])] = $user;
+                    }
+                } else {
+                    $coDeps[strToLower($user->getFirstname())] = $user;
```

Comment:
There is a danger of this being overwritten and codeps going missing. Consider 2 co deps with the same first name. Maybe add a unique identifier?

Reply:
yeah of course - duh